tests/prague/eip7702_set_code_tx/test_invalid_tx.py

```diff
@@ -142,13 +166,77 @@ def test_invalid_tx_invalid_auth_chain_id(
     )
 
 
+@pytest.mark.parametrize(
+    "auth_chain_id",
+    [pytest.param(0), pytest.param(1)],
+)
+@pytest.mark.parametrize(
+    "delegate_address",
+    [
+        pytest.param(Spec.RESET_DELEGATION_ADDRESS, id="reset_delegation_address"),
+        pytest.param(Address(1), id="non_zero_address"),
+    ],
+)
+def test_invalid_tx_invalid_auth_chain_id_encoding(
+    transaction_test: TransactionTestFiller,
+    pre: Alloc,
+    delegate_address: Address,
+    auth_chain_id: int,
+):
+    """
+    Test sending a transaction where the chain id field of an authorization has an incorrect
+    encoding.
+    """
+
+    class ModifiedAuthorizationTuple(AuthorizationTuple):
+        chain_id: OversizedInt  # type: ignore
+
+        def to_list(self) -> List[Any]:
+            """Return authorization tuple as a list of serializable elements."""
+            return [
+                self.chain_id,
+                self.address,
+                Uint(self.nonce),
+                Uint(self.v),
+                Uint(self.r),
+                Uint(self.s),
+            ]
+
+        def to_signing_list(self) -> List[Any]:
```

Comment:
Maybe the ones with `to_signing_list` modifications could be parametrized to sign the correct thing or the morphed rlp.